### PR TITLE
[backup] trait ErrorNotes 

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

shortcut to log some context info and the stack trace when error happens.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/master/CONTRIBUTING.md#pull-requests)?

Y
## Test Plan

existing coverage.

tried some bad commands to observe the error log.

### CODE REVIEW

File: storage/backup/backup-cli/src/utils/trace_err.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+use std::fmt::Display;
+
+pub(crate) trait TraceError<T, E: Display> {
+    fn trace_err(self, notes: &str) -> Result<T, E>;
+}
+
+impl<T, E: Display> TraceError<T, E> for Result<T, E> {
+    fn trace_err(self, notes: &str) -> Result<T, E> {
+        if let Err(e) = &self {
+            error!(error = %e, notes = notes, "Error traced.");
```
Comment (bmwill) [inline storage/backup/backup-cli/src/utils/trace_err.rs:14]: one word of caution here is that this is the only line number that will show up for these errors (I mean you'll have some form of a stack trace as well)

Comment (msmouse) [reply storage/backup/backup-cli/src/utils/trace_err.rs:14]: Ugh. I see. Thanks!
The `notes` will definitely help, and hopefully the stack I can relate it to the caller even in an async environment.

Comment (msmouse) [reply storage/backup/backup-cli/src/utils/trace_err.rs]: K, added file!() and line!() as context when necessary.

**Review by lightmark — APPROVED** [review]

**Pull request merged**